source/backend/core/Http/OpenIdConnectRequestClient.cs

```diff
@@ -105,24 +105,22 @@ public OpenIdConnectRequestClient(
         /// <returns></returns>
         public async Task<string> RequestAccessToken()
         {
-            HttpResponseMessage response;
-            if (_accessToken == null || string.IsNullOrWhiteSpace(_accessToken.AccessToken) || (!string.IsNullOrWhiteSpace(_accessToken.RefreshToken) && _tokenHandler.ReadJwtToken(_accessToken.RefreshToken).ValidTo <= DateTime.UtcNow))
+            if (IsValidToken())
             {
-                // If there is no access token, or the refresh token has expired.
-                response = await RequestToken();
+                // We have a valid token, keep on using it.
+                return $"Bearer {_accessToken.AccessToken}";
             }
-            else if (!string.IsNullOrWhiteSpace(_accessToken.AccessToken)
-                && _tokenHandler.ReadJwtToken(_accessToken.AccessToken).ValidTo <= DateTime.UtcNow
-                && !string.IsNullOrWhiteSpace(_accessToken.RefreshToken)
-                && _tokenHandler.ReadJwtToken(_accessToken.RefreshToken).ValidTo > DateTime.UtcNow)
+
```

Comment:
Testing locally, I did not get a refresh token. So this section was not being called